### PR TITLE
[ADD] base_partner_validate_address: Abstract Address Validations

### PR DESCRIPTION
**Pull request by lasley**
This module provides an abstract core allowing for address verifications via external providers. It is a work in progress, and will likely be adjusted as I implement in our three modules which all have address validation.

It uses the `base_external_system` interface/adapter mechanism so that no views are actually required for the individual systems. I still need to add a button in the partner to activate the wizard, but that's a piece of cake.

### CODE REVIEW

File: env/include/site/python2.7/greenlet/greenlet.h [hunk 1]
```diff
@@ -0,0 +1,148 @@
+/* vim:set noet ts=8 sw=8 : */
```
Comment (lasley) [inline env/include/site/python2.7/greenlet/greenlet.h]: ref #503

**Comment by lasley** [comment]
@obulkin please take a look at this when you get a chance

File: base_partner_validate_address/README.rst [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+   :alt: License: LGPL-3
+
+=============================
+Base Partner Validate Address
+=============================
+
+Adds an abstract core for address verification via remote interfaces.
+
+Installation
+============
```
Comment (obulkin) [inline base_partner_validate_address/README.rst]: There are some empty sections here that should be removed.

File: base_partner_validate_address/wizards/wizard_address_validate.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    country_id_original = fields.Many2one(
+        string='Country',
+        comodel_name='res.country',
+        related='partner_id.country_id',
+        readonly=True,
+    )
+    latitude_original = fields.Float(
+        string='Latitude',
+        related='partner_id.latitude',
+    )
```
Comment (obulkin) [inline base_partner_validate_address/wizards/wizard_address_validate.py]: Shouldn't these last two fields also be read-only?

**Review by obulkin — CHANGES_REQUESTED** [review]
Generally, this looks good, but the latitude and longitude fields introduced in the module aren't actually updated by the validation logic. It seems like they should either be dropped or included in the validation process.

**Comment by lasley** [comment]
Thanks @obulkin - comments attended to

File: base_partner_validate_address/models/address_validate.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    def get_address(self, partner):
+        """Returns an address suggestion from the interface.
+
+        Args:
+            partner (ResPartner): Partner record to get address for.
+
+        Returns:
+            dict: The suggested address. Should contain keys: ``street``,
+                ``street2``, ``city``, ``state_id``, ``zip``, ``country_id``,
+                ``is_valid``, and ``validation_messages``.
```
Comment (obulkin) [inline base_partner_validate_address/models/address_validate.py:62]: Not a blocker, but now that latitude and longitude are fully supported, that should also be mentioned here.

**Review by obulkin — APPROVED** [review]

File: base_partner_validate_address/__manifest__.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+# Copyright 2016-2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+{
+    'name': 'Base Partner Validate Address',
+    'summary': 'Provides an abstract interface/adapter mechanism allowing '
+               'for the unification of address verification providers.',
+    'version': '10.0.1.0.0',
+    'category': 'Extra Tools',
+    'website': 'https://laslabs.com/',
```
Comment (NL66278) [inline base_partner_validate_address/__manifest__.py:11]: Should be https://github.com/oca/partner-contact

File: base_partner_validate_address/__init__.py [hunk 6]
```diff
@@ -0,0 +1,6 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016-2017 LasLabs Inc.
```
Comment (NL66278) [inline base_partner_validate_address/__init__.py:2]: Remove copyright lines from init files

File: base_partner_validate_address/__init__.py [hunk 7]
```diff
@@ -0,0 +1,6 @@
...
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
```
Comment (NL66278) [inline base_partner_validate_address/__init__.py:3]: Make license lines https:

**Review by NL66278 — CHANGES_REQUESTED** [review]
:+1: Interesting module. Few changes would be good, see other comments.

**Comment by NL66278** [comment]
@lasley Will you be taking up tehe comments on this PR?

**Comment by rousseldenis** [comment]
@NL66278 , @lasley is no more present or developing on ODOO nor OCA. I think we can close this.

**Pull request closed without merging**